### ISSUE TITLE
SEO app makes page title [object Object]

### ISSUE THREAD

**Issue by ComLock** [enonic/app-metafields#27]
Our page controller has the following in thymeleaf for <head><title>:

```
<title data-th-if="${not decorator}">Posten.no</title>
<title data-th-if="${decorator}">$title</title>
```

The SEO configuration for the page is:
```
"com-enonic-app-metafields": {
      "meta-data": {
        "seoDescription": "Når du er borte i en begrenset periode kan du få posten sendt til ditt midlertidige oppholdssted.",
        "blockRobots": false
      }
    }
```

When SEO app 1.3.2 is running view source on page gives (regardless of wether decorator is true or false):

```
<title>[object Object]</title>
```

Stopping the seo app gives 500:
```
com.enonic.xp.resource.ResourceNotFoundException: Resource [com.enonic.app.metafields:/site/filters/add-metadata.js] was not found (java.lang.RuntimeException)
```

Uninstalling the SEO app gives title as programmed in the template.

**Comment by ComLock** [comment]
In production we are using SEO-1.1.2 which doesn't have the same bug. However it also overrides title to the displayName of the page. Regardless of decorator = true or false

**Comment by ComLock** [comment]
Looking at the SEO siteConfig it doesn't appear possible to NOT override the title element. I will have a look at the SEO filter code now.

**Comment by ComLock** [comment]
Hmm, lots of things in that code.

For example line 79 is redundant as line 72 has already done it without reordering elements in head.
https://github.com/enonic/app-metafields/blob/master/src/main/resources/site/filters/add-metadata.js#L72-L79

**Comment by ComLock** [comment]
The filter code simply overrides the title element.
This means we cannot use it as it collides with our decorator code.

**Comment by ComLock** [comment]
Note keep in mind the filter received the rendered version of the thymeleaf, so it will only have a single title element.

**Comment by ComLock** [comment]
Even though I don't like the fact the SEO app simply overrides the title element, I guess it's ok.
Because all I need to do is override it once more. So I must make a decorator filter that runs after the SEO filter.

**Comment by ComLock** [comment]
So the decorator filter fixes the instance when decorator=true.

We still get [object Object] when decorator=false

**Comment by ComLock** [comment]
The problem is where the title is in the content-type config:
content.data.heading.title

I've tried setting the siteConfig.pathsTitles to
- content.data.heading.title
- data.heading.title
- heading.title
- title

No luck

**Comment by ComLock** [comment]
This does not handle nested variables:
https://github.com/enonic/app-metafields/blob/master/src/main/resources/lib/site.js#L29

**Comment by ComLock** [comment]
The problem got introduced in 1.2.0

**Comment by Bellfalasch** [comment]
Fixed in 1.3.3
